services/classic/management/http.go

```diff
@@ -161,6 +157,10 @@ func (client client) sendRequest(httpClient *http.Client, url, requestType, cont
 				if numberOfRetries == 0 {
 					return nil, azureErr
 				}
+				if response.StatusCode == http.StatusServiceUnavailable {
```

Comment:
I'm not super-familiar with ASM but for ARM we have a list of standard HTTP status codes to automatically retry on, you can find it [here](https://github.com/Azure/go-autorest/blob/master/autorest/client.go#L48).  Up to you if you want to expand this or not.

Reply:
ASM has some funny behavior. 
I initially went with 429 (StatusTooManyRequests) as one would expect from a throttling message. But ASM does not seem to return that in this scenario.